android/engine/src/main/java/org/smartregister/fhircore/engine/ui/theme/Themes.kt

```diff
@@ -16,16 +16,59 @@
 
 package org.smartregister.fhircore.engine.ui.theme
 
+import androidx.appcompat.app.AppCompatDelegate
 import androidx.compose.material.MaterialTheme
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
 
 /**
  * TODO fix issue with ktfmt formatting annotated high order functions. Current workaround below:
  * lambda in this format content: (@Composable() () -> Unit) to allow spotlessApply
  *
  * To enable app theme set darkTheme = isSystemInDarkTheme
  */
+//@Composable
+//fun AppTheme(darkTheme: Boolean = false, content: (@Composable() () -> Unit)) {
+//  MaterialTheme(colors = if (darkTheme) DarkColors else LightColors, content = content)
+//}
+
+class ThemeViewModel : ViewModel() {
+  var isSystemInDarkTheme by mutableStateOf(false)
+    private set
+
+  fun updateTheme(isDarkTheme: Boolean) {
+    isSystemInDarkTheme = isDarkTheme
+    val theme = if (isSystemInDarkTheme) AppCompatDelegate.MODE_NIGHT_YES else AppCompatDelegate.MODE_NIGHT_NO
+    AppCompatDelegate.setDefaultNightMode(theme)
+  }
+}
+
 @Composable
-fun AppTheme(darkTheme: Boolean = false, content: (@Composable() () -> Unit)) {
-  MaterialTheme(colors = if (darkTheme) DarkColors else LightColors, content = content)
+fun AppTheme(content: @Composable () -> Unit) {
```

Comment:
Are you able to run spotlessCheck is failing because of this `@Composable () -> Unit`. Revert to this workaround `@Composable () -> Unit`. It's a know spotlessCheck issue.

Reply:
Not sure it has been resolved in newer versions of `Spotless` plugin